### PR TITLE
Chat menu

### PR DESCRIPTION
**Pull request by iqmulator**

### CODE REVIEW

**Comment by iqmulator** [comment]
image
image
image
image
image
image
image
image
image
image

**Comment by iqmulator** [comment]
image

File: src/dotnet/Chat.UI.Blazor/Components/ChatsTab.razor [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    private string GetCss(ChatGroup group) {
+        return group switch
+        {
+            ChatGroup.Pinned => "pinned-chats",
+            ChatGroup.Unpinned => "unpinned-chats custom-scrollbar",
+            _ => "",
+            };
+    }
+
+    private bool IsListening(string chatId) {
```
Comment (crui3er) [inline src/dotnet/Chat.UI.Blazor/Components/ChatsTab.razor:74]: It seems method 'IsListening' is not used.

File: src/dotnet/UI.Blazor/Components/ContextMenu/context-menu.ts [hunk 2]
```diff
@@ -65,7 +65,8 @@ export class ContextMenu {
     }
 
     private updatePosition(menu: HTMLDivElement) {
-        const maxBottom = document.documentElement.clientHeight - 90
+        const editor = document.querySelector('.chat-message-editor');
```
Comment (crui3er) [inline src/dotnet/UI.Blazor/Components/ContextMenu/context-menu.ts:68]: What is relation between chat-message-editor and context menu? Why does context menu know about chat-message-editor?

**Review by crui3er — CHANGES_REQUESTED** [review]

File: src/dotnet/Chat.UI.Blazor/Components/ChatMenu/ChatMenuPlayback.razor [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+@{
+    var m = State.ValueOrDefault ?? Model.None;
+    var playingHistoricalClass = m.IsPlayingHistorical ? " opacity-60" : "";
+    var cls = "play " + (m.IsPlayingActive ? "on" : "off") + playingHistoricalClass;
+    var isListening = m.IsPlayingActive || m.IsPlayingHistorical;
+}
+
+@if (!isListening) {
+    <MenuItem Title="Listen" Name="Listen Chat" Click="@(_ => ChatUI.IsPlaying.Value = !m.IsPlayingActive)" Class="@cls">
```
Comment (alexis-kochetov) [inline src/dotnet/Chat.UI.Blazor/Components/ChatMenu/ChatMenuPlayback.razor:12]: Let's agree to avoid any logic at markup - please extract to a method

Comment (alexis-kochetov) [reply src/dotnet/Chat.UI.Blazor/Components/ChatMenu/ChatMenuPlayback.razor:12]: I'm not sure that you change playback state there.

**Review by alexis-kochetov — CHANGES_REQUESTED** [review]

**Pull request merged**